workflows/src/main/kotlin/com/r3/corda/lib/accounts/workflows/flows/RequestAccountInfoByNameFlows.kt

```diff
@@ -0,0 +1,74 @@
+package com.r3.corda.lib.accounts.workflows.flows
+
+import co.paralleluniverse.fibers.Suspendable
+import com.r3.corda.lib.accounts.contracts.states.AccountInfo
+import com.r3.corda.lib.accounts.workflows.internal.accountService
+import net.corda.core.contracts.StateAndRef
+import net.corda.core.flows.*
+import net.corda.core.identity.Party
+import net.corda.core.node.StatesToRecord
+import net.corda.core.utilities.unwrap
+
+/**
+ * This flow can be used to check whether an account is hosted by a particular host by passing account name. If it is then the host will share
+ * the account info with the requester. This flow should be used in a situation where UUID is difficult to obtain for a given
+ * account. For e.g: a node can create accounts for its employees using SSN/NIN or employeeId
+ *
+ * @property accountName account name to request the [AccountInfo] for hosted at [host] node.
+ * @property host session to request the [AccountInfo] from.
+ */
+class RequestAccountInfoByNameFlow(private val accountName: String, val host: FlowSession) : FlowLogic<AccountInfo?>() {
+    @Suspendable
+    override fun call(): AccountInfo? {

```

Comment:
If `host.counterParty == ourIdentity` then just fetch the account from the current node.